examples/text-classification/run_glue.py

```diff
@@ -85,6 +86,9 @@ def main():
             f"Output directory ({training_args.output_dir}) already exists and is not empty. Use --overwrite_output_dir to overcome."
         )
 
+    if "cuda" not in str(training_args.device) or not is_apex_available():
```

Comment:
Can we just change the test code? Or is this fixing a real bug?

Reply:
@sshleifer I wasn't sure whether this is a good solution but I have added this for certain reasons.
1.  For the apex to work we need to have cuda. So when was implementing this I had a scenario where the apex was installed but the GPU is not enabled thus I added both the checks. This checks for the cuda first before checking if apex is available or not.
---
2. I wasn't sure if the `training_args.device` will give `cuda` or `cuda:0` when there are multiple GPUs thus I have added a check to know if the `training_args.device` has a cuda. 
---
> Did you also add multi GPU support? If so you need to test it on a machine/show that it works, and it should probably be a separate PR.
3. I didn't find a way to test this on multiple GPU. I will try my best to find a way to do this. But in `examples/text-generation/run_generation.py` I have added `local_rank` because I need to make every test file to have consistent arguments. This doesn't mean it adds a multi GPU support, I can remove this as it is confusing.

I am pretty sure this might be a wrong implementation please correct me if any of the above-mentioned points are not correct and if there is any other best alternative to do this.

Reply:
Lets check if cuda is available in `test_examples.py`, and if it is pass the fp16 argument.
Otherwise, don't pass the fp16 argument.